packages/react-native/ReactCommon/react/nativemodule/core/platform/android/ReactCommon/JavaTurboModule.cpp

```diff
@@ -56,16 +56,46 @@ JavaTurboModule::~JavaTurboModule() {
 
 namespace {
 
+constexpr auto kReactBuildConfigJavaDescriptor = "com/facebook/react/common/build/ReactBuildConfig";
+
+bool getReactBuildConfigBoolValue(const char *name) {
+    static const auto reactBuildConfigClass = facebook::jni::findClassStatic(kReactBuildConfigJavaDescriptor);
+    const auto field = reactBuildConfigClass->getStaticField<jboolean>(name);
+    return reactBuildConfigClass->getStaticFieldValue(field);
+}
+
+constexpr auto kReactFeatureFlagsJavaDescriptor = "com/facebook/react/config/ReactFeatureFlags";
+
+bool getFeatureFlagBoolValue(const char *name) {
+    static const auto reactFeatureFlagsClass = facebook::jni::findClassStatic(kReactFeatureFlagsJavaDescriptor);
+    const auto field = reactFeatureFlagsClass->getStaticField<jboolean>(name);
+    return reactFeatureFlagsClass->getStaticFieldValue(field);
+}
+
+bool traceTurboModulePromiseRejections() {
+  static bool traceRejections = getFeatureFlagBoolValue("traceTurboModulePromiseRejections");
+  static bool isDebugMode = getReactBuildConfigBoolValue("DEBUG");
+  return isDebugMode || traceRejections;
+}
+
+bool rejectTurboModulePromiseOnNativeError() {
+    static bool rejectOnError = getFeatureFlagBoolValue("rejectTurboModulePromiseOnNativeError");
+    return rejectOnError;
+}
+
 struct JNIArgs {
   JNIArgs(size_t count) : args_(count) {}
   std::vector<jvalue> args_;
   std::vector<jobject> globalRefs_;
 };
 
+using CallbackExecutor = std::function<void(jsi::Runtime&, jsi::Function&, const std::vector<jsi::Value>&)>;
+
 jni::local_ref<JCxxCallbackImpl::JavaPart> createJavaCallbackFromJSIFunction(
-    jsi::Function&& function,
-    jsi::Runtime& rt,
-    const std::shared_ptr<CallInvoker>& jsInvoker) {
+    jsi::Function &&function,
+    jsi::Runtime &rt,
+    const std::shared_ptr<CallInvoker> &jsInvoker,
+    CallbackExecutor&& callbackExecutor) {
```

Comment:
Please rebase on https://github.com/facebook/react-native/pull/39952/ which should simplify this code somewhat.

Reply:
The commit was reverted. I guess I can ignore this. https://github.com/facebook/react-native/pull/40951

Reply:
Unfortunately we found some issues with AsyncCallback which should have now been fixed and I'm planning to merge this again (https://github.com/facebook/react-native/pull/41048)

Reply:
Thanks for the update. I've tried to use the AsyncCallback, but I'm getting only `undefined` back in JS.

Do you see any obvious issue with the way I use it?

https://github.com/facebook/react-native/pull/37484/files#diff-3af3932eff18aefc435b8fb98f6feebbb52809dec07005e937909c8f0d850cb1R511-R522

Reply:
Got it thanks, I used it now.